designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,54 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace 
```

Comment:
also, I think that there's an important distinction between technical limitations and facilitating an organization approach by customers; I really like this idea from the organizational approach but don't see technical limitations.

Reply:
The main technical limitation I could anticipate is in the clusterctl move command during a cluster upgrade. I ran an experiment with 3 workload clusters (1 etcd, 1 CP, 1 MD) on a management cluster (3 etcd, 4 CP, 3 MD) and observed that 188 resources were moved. Each resource is probably only a few KB, but if we increase that by one or two orders of magnitude, I am concerned about the API load on a bootstrap cluster running in a smaller machine somewhere. Does that seem valid to you? @danbudris